src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -210,7 +211,22 @@ private T Deserialize<T>(JsonSerializer serializer, JToken jToken)
 
         private JsonSerializer GetPayloadSerializer(int? version)
         {
-            return version == 2 ? payloadSerializer2 : payloadSerializer;
+            if (version == null)
+            {
+                version = 1;
```

Comment:
Why do you make version = 1, if it is null, why not 2?

Reply:
Because 1 is the baseline at which all versions of test.sdk can communicate, 2 is optimization. So we ask over version 1: 

Console: Hey testhost, what is the highest version of protocol you can use? I can do 3.
Testhost: I can do 1. 
Console: Okay we will use 1. 

Or:

Console: Hey testhost, what is the highest version of protocol you can use? I can do 2.
Testhost: I can do 4. 
Console: Okay we will use 2.